inference-engine/tests/unit/cpu/nodes/mkldnn_reorder_node.cpp

```diff
@@ -0,0 +1,144 @@
+    auto outputNode = std::make_shared<MKLDNNPlugin::MKLDNNInputNode>(MKLDNNPlugin::Shape(dstDims),
+                                                                       prec,
+                                                                       "Reorder_Output", "Output",
+                                                                       cpuEngine, weightsCache);
+
+    auto parentEdge = std::make_shared<MKLDNNPlugin::MKLDNNEdge>(inputNode, reorderNode, 0, 0);
+    auto childEdge = std::make_shared<MKLDNNPlugin::MKLDNNEdge>(reorderNode, outputNode, 0, 0);
+    parentEdge->changeStatus(MKLDNNPlugin::MKLDNNEdge::Status::NeedAllocation);
+    childEdge->changeStatus(MKLDNNPlugin::MKLDNNEdge::Status::NeedAllocation);
+    reorderNode->addEdge(parentEdge);
+    reorderNode->addEdge(childEdge);
+
+    const std::vector<size_t> srcBlockedDims = getBlockedDims(srcDims, srcOrder);
+    const std::vector<size_t> srcStrides = getStrides(srcBlockedDims);
+    const std::vector<size_t> offsetPaddingToData(srcDims.size(), 0);
+
+    const std::vector<size_t> dstBlockedDims = getBlockedDims(dstDims, dstOrder);
+    const std::vector<size_t> dstStrides = getStrides(dstBlockedDims);
+
+    const MKLDNNPlugin::BlockedMemoryDesc inputDesc(prec, srcDims, srcBlockedDims,
+                                                     srcOrder, 0, offsetPaddingToData, srcStrides);
+
+    const MKLDNNPlugin::BlockedMemoryDesc outputDesc(prec, srcDims, getBlockedDims(srcDims, dstOrder),
+                                                      dstOrder, 0, offsetPaddingToData, dstStrides);
+
+    auto parent_memory = std::make_shared<MKLDNNPlugin::MKLDNNMemory>(cpuEngine);
+    auto childMemory = std::make_shared<MKLDNNPlugin::MKLDNNMemory>(cpuEngine);
+    parent_memory->Create(inputDesc, srcData);
+    childMemory->Create(outputDesc, dstData);
+    parentEdge->reuse(parent_memory);
+    childEdge->reuse(childMemory);
+
+    reorderNode->setDescs(inputDesc, outputDesc);
+    std::vector<std::shared_ptr<MKLDNNPlugin::MKLDNNNode>> nodes {inputNode, reorderNode, outputNode};
+    for (auto &n : nodes) {
+        n->init();
+        n->getSupportedDescriptors();
+        n->initSupportedPrimitiveDescriptors();
+        n->selectPrimitiveDescriptorByIndex(0);
+    }
+    auto config = outputNode->getSelectedPrimitiveDescriptor()->getConfig();
+    config.inConfs.resize(1);
+    config.inConfs[0].inPlace = 1;
+    outputNode->getSelectedPrimitiveDescriptor()->setConfig(config);
+    reorderNode->createPrimitive();
+
+    mkldnn::stream strm(cpuEngine);
+    reorderNode->execute(strm);
+    return;
+}
+}// namespace
+/*
+ * The tests check that dst batch strides are correctly taken into account by
+ * MKLDNNReorderNode::optimizedNcsp2Nspc() and MKLDNNReorderNode::optimizedNspc2Ncsp().
+ * This represents the case when the reorder is followed by an inplace concat.
+ */
+TEST(InplaceChildReorderTest, NSPC2NCSP) {
```

Comment:
may i ask you also add test for not inplace `optimizedNcsp2Nspc` and `optimizedNspc2Ncsp` reorder primitive?

Reply:
Yes, it seems logical. Covered the not-inplace cases, but had to rewrite the tests to avoid copy-paste. Now using fixtures and parameterized tests which is more extensible.